node/src/components/contract_runtime/rewards.rs

```diff
@@ -495,16 +504,43 @@ pub(crate) fn rewards_for_era(
                     .checked_mul(&rewards_info.reward(signed_block_era)?)
                     .ok_or(RewardsError::ArithmeticOverflow)?;
 
-                increase_value_for_key(signing_validator, contribution_reward)?;
-                increase_value_for_key(block.proposer.clone(), collection_reward)?;
+                increase_value_for_key_and_era(
+                    signing_validator,
+                    signed_block_era,
+                    contribution_reward,
+                )?;
+                increase_value_for_key_and_era(
+                    block.proposer.clone(),
+                    signed_block_era,
+                    collection_reward,
+                )?;
             }
         }
     }
 
+    let rewards_map_to_vec = |rewards_map: BTreeMap<EraId, Ratio<U512>>| {
+        let min_era = rewards_map
+            .iter()
+            .find(|(_era, &amount)| !amount.numer().is_zero())
+            .map(|(era, _amount)| era)
+            .copied()
+            .unwrap_or(current_era_id);
+        EraId::iter_range_inclusive(min_era, current_era_id)
+            .rev()
```

Comment:
Is there a particular reason to reverse the order of iteration here?

Reply:
It's just so that the latest era is the first value in the vector of reward amounts. The idea was to avoid having unnecessary zeros for the previous era if a validator is only rewarded in the current era, so then a single value must mean an amount for the current era, and values further in the vector (up to 1 other value currently) would be for previous eras.